apps/web/pages/api/auth/[...nextauth].tsx

```diff
@@ -499,7 +526,12 @@ export default NextAuth({
               },
             });
 
-            return true;
+            if (existingUserWithEmail.twoFactorEnabled) {
+              const jwt = await signJwt({ email: existingUserWithEmail.email });
+              return `/auth/login?totp=${jwt}`;
```

Comment:
This is the redirection needed to let the user input a totp code in an external provider use case.

Reply:
This should be in a separate function 🙏 as not to repeat yourself